frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-ref.ts

```diff
@@ -0,0 +1,111 @@
+import { K8sModel } from '../../api/common-types';
+import {
+  GetAPIVersionForModel,
+  GetGroupVersionKindForResource,
+  GetGroupVersionKindForModel,
+} from '../../api/k8s-types';
+import { K8sGroupVersionKind, K8sResourceKindReference } from '../../extensions/console-types';
+
+/**
+ * @deprecated - This will become obsolete when we move away from K8sResourceKindReference to K8sGroupVersionKind
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s resource.
+ * @param K8sGroupVersionKind Pass K8sGroupVersionKind which will have group, version, and kind of a k8s resource.
+ * @param K8sGroupVersionKind.group Pass group of k8s resource or model.
+ * @param K8sGroupVersionKind.version Pass version of k8s resource or model.
+ * @param K8sGroupVersionKind.kind Pass kind of k8s resource or model.
+ * @return The reference for any k8s resource i.e `group~version~kind`.
+ * If the group will not be present then "core" will be returned as part of the group in reference.
+ * * */
+export const getReference = ({
```

Comment:
Only question I have would be, did this get added part of this PR or was it moved from somewhere else? If we can avoid adding (new) deprecated items to the public API I think think that would be best. If this was already exposed and I'm just blind (I can't find it), then ignore away 🙂

Reply:
APIs outside of the src/apis folder are not public. This isn't clearly documented, though. I worry people are going to use the internal APIs not realizing they're internal.

I believe there is a way to limit what's exposed in webpack to avoid this... It might require webpack 5. @vojtechszocs @christianvogt do you know?

Reply:
Ah doh... I forgot we wrote internally then exposed optionally in the `core.ts`. My bad.